### ISSUE TITLE
Multitasking

### ISSUE THREAD

**Issue by SamTebbs33** [SamTebbs33/pluto#35]
We'll need a rudimentary cooperative multitasking system that presents an interface to a program, allowing it to have its state stored, relinquish control to another task. When control is returned to the task in the future, its state should be restored.

Control could be relinquished by sending an interrupt since the interrupt handling system already does a lot of the work of saving the CPU state, and this would lay the foundation for a more advanced multitasking and scheduling system, where all task transitions occur via syscalls and timer interrupts.

Preemptive scheduling with a timer interrupt should probably be done in a separate issue.

Depends on #36 if interrupts are the only way for a program to relinquish control.

**Comment by SamTebbs33** [comment]
Closed by #138
